unit/memory-analyzer/gdb_api.cpp

```diff
@@ -104,94 +104,111 @@ void gdb_api_internals_test()
 TEST_CASE("gdb api internals test", "[core][memory-analyzer]")
 {
 #ifdef RUN_GDB_API_TESTS
-  gdb_api_internals_test();
+  bool has_gdb = system("gdb --version") == 0;
+  SECTION("check gdb is on the PATH")
+  {
+    REQUIRE(has_gdb);
+  }
+  if(has_gdb)
+  {
+    gdb_api_internals_test();
+  }
 #endif
 }
 
 TEST_CASE("gdb api test", "[core][memory-analyzer]")
 {
 #ifdef RUN_GDB_API_TESTS
-  compile_test_file();
-
+  bool has_gdb = system("gdb --version") == 0;
+  SECTION("check gdb is on the PATH")
   {
-    gdb_apit gdb_api("test", true);
-    gdb_api.create_gdb_process();
+    REQUIRE(has_gdb);
+  }
+  if(has_gdb)
+  {
+    compile_test_file();
 
-    try
     {
-      const bool r = gdb_api.run_gdb_to_breakpoint("checkpoint");
-      REQUIRE(r);
-    }
-    catch(const gdb_interaction_exceptiont &e)
-    {
-      std::cerr << "warning: cannot fully unit test GDB API as program cannot "
-                << "be run with gdb\n";
-      std::cerr << "warning: this may be due to not having the required "
-                << "permissions (e.g., to invoke ptrace() or to disable ASLR)"
-                << "\n";
-      std::cerr << "gdb_interaction_exceptiont:" << '\n';
-      std::cerr << e.what() << '\n';
-
-      std::ifstream file("gdb.txt");
-      CHECK_RETURN(file.is_open());
-      std::string line;
-
-      std::cerr << "=== gdb log begin ===\n";
+      gdb_apit gdb_api("test", true);
+      gdb_api.create_gdb_process();
 
-      while(getline(file, line))
+      try
       {
-        std::cerr << line << '\n';
+        const bool r = gdb_api.run_gdb_to_breakpoint("checkpoint");
+        REQUIRE(r);
       }
+      catch(const gdb_interaction_exceptiont &e)
+      {
+        std::cerr
```

Comment:
Don't we use message handlers in unit tests?

Reply:
At least not consistently, I found a few other tests that call `std:cerr`. Should I modify this one?

Reply:
@peterschrammel What value would the use of message handlers add here?